src/PanelTraits/FakeFields.php

```diff
@@ -66,4 +69,16 @@ public function compactFakeFields($request, $form = 'create', $id = false)
         // since no modifications or additions have been made to $request
         return $request;
     }
+
+    /**
+     * @return array
+     */
+    protected function getModelCasts()
+    {
+        $modelReflection = new \ReflectionClass($this->model);
+        $castsAttribute = $modelReflection->getProperty('casts');
+        $castsAttribute->setAccessible(true);
+
+        return $castsAttribute->getValue($this->model);
```

Comment:
This reads less surprisingly as:

```
    $castedFields = $castsAttribute->getValue($this->model);
    return $castedFields;
```

I know that it's not necessary to split those two out but I'm sitting here wondering, "Hmmm, why does something that **looks like** a plural suddenly just get a single value?" At least breaking it into two makes it more obvious and PHP should be sane enough to optimize the double step out.